roombasim/pittras/task/hold_position_task.py

```diff
@@ -0,0 +1,45 @@
+'''
+hold_position_task.py
+'''
+
+import numpy as np
+
+from roombasim.ai import Task
+
+class HoldPositionTaskStates:
+    '''
+    States for the Hold Position task.
+    '''
+    init = 0
+    holding = 1
+    done = 2
+
+class HoldPositionTask(Task):
+    '''
+    A task to hold the drone at its current absolute position for some duration
+    of time. The task accepts a float64 that specifies such duration in seconds,
+    which means holding the current position indefinitely if less than or equal
+    to zero.
+    '''
+
+    def __init__(self, hold_duration):
+        self.hold_duration = hold_duration
+        self.state = HoldPositionTaskStates.init
+        self.start_time = 0
+
+    def update(self, delta, elapsed, state_controller, environment):
+        if self.done == HoldPosition.done:
+            # TODO(zacyu): Signal the termination of the task when such
+            #              capability is added to the framework.
+            return
```

Comment:
Sounds good. @hgarrereyn what do you think?

Reply:
The reason we do it that way in the motion planner is for safety reasons. The motion planner is stable software and we trust it to check whether requests from tasks are reasonable, while the tasks are unstable and not trusted. So, the tasks hand all their concerns to the notion planner to be executed. That isn't an issue here, do I think it's easier to just call the `control` method inside the tasks

Reply:
It might actually make sense to mirror the way `get_desired_command` returns state/command values. That way we could abstract away the actual "control" mechanism so that it would be easier to swap real-world/simulator if we want to do that.

Reply:
We shouldn't actually have to implement real-world code for this though, because it already exists in the motion planner. The layer we're swapping should be above this, right?